packages/grafana-data/src/types/datasource.ts

```diff
@@ -362,6 +362,12 @@ abstract class DataSourceApi<
    * @alpha -- experimental
    */
   getDefaultQuery?(app: CoreApp): Partial<TQuery>;
+
+  /**
+   * Define if the contents of a query are complete
+   * @alpha -- experimental
+   */
+  isCompleteQuery?(query?: DataQuery): boolean;
```

Comment:
I am wondering if `isCompleteQuery` is the right name, cause `complete query !== non-empty query`. For example in Loki code, we are checking if `query.expr` is not empty. But if the `query.expr` is for example `abc`, I wouldn't consider it complete query. Have you considered other names for this method?